src/Compilers/CSharp/Portable/Symbols/ParameterSymbol.cs

```diff
@@ -415,5 +415,7 @@ protected override ISymbol CreateISymbol()
         {
             return new PublicModel.ParameterSymbol(this);
         }
+
+        internal virtual bool HasEnumeratorCancellationAttribute => false;
```

Comment:
>internal virtual bool HasEnumeratorCancellationAttribute => false; [](start = 8, length = 66)

We prefer to add abstract APIs to symbols instead. This way we can make sure all leaf classes properly implement the API and it is much easier to confirm that all scenarios are also tested. For example, it doesn't look like PEParameterSymbol implements this API properly, i.e. it always returns false, which is not what I would expect. It looks like the same issue exists for retargeting and substituted parameters. If this API doesn't make sense for them, then it probably shouldn't be added to the root base type, but instead should be moved to the more specialized derived type.